app/models/salt_job.rb

```diff
@@ -0,0 +1,34 @@
+# Store selected salt job ids, and record on their result
+class SaltJob < ActiveRecord::Base
+  validates :jid, uniqueness: true
+
+  scope :all_open, -> { where(retcode: nil) }
+  scope :failed, -> { where.not(retcode: [nil, 0]) }
+  scope :jids, -> { pluck(:jid) }
+
+  def complete!(retcode = 0, master_trace: nil, minion_trace: nil)
+    changes = { retcode: retcode }
+    changes[:master_trace] = master_trace if master_trace
+    changes[:minion_trace] = minion_trace if minion_trace
+    update_attributes! changes
+
+    parse_upstream_error if failed?
+  end
+
+  def completed?
+    retcode.present?
+  end
+
+  def succeeded?
+    retcode.present? && retcode.zero?
```

Comment:
you could use `completed?` here (and below in `def failed?`)

Reply:
Resolved in 7b9ecda4fadd37f3ff15a01f80399ee70c656fbd